### ISSUE TITLE
AutoencoderKL decoder implementation is missing the trainable logvar scalar from compvis codebase

### ISSUE THREAD

**Issue by AmericanPresidentJimmyCarter** [huggingface/diffusers#5731]
### Describe the bug

This is necessary for training/finetuning of the VAE. Please see the original implementation from compvis over at:

https://github.com/CompVis/latent-diffusion/blob/main/ldm/modules/losses/contperceptual.py#L20

### Reproduction

The trainable logvar scalar does not exist in your implementation.

### System Info

diffusers==main

### Who can help?

@sayakpaul @yiyixuxu

**Comment by yiyixuxu** [comment]
I think we can support this 

wtdyt? @sayakpaul @patrickvonplaten @DN6

**Comment by sayakpaul** [comment]
This does look like a simple change and won't be backwards compatible I think. So, I am okay.

**Comment by AmericanPresidentJimmyCarter** [comment]
I was talking to @crowsonkb about this and we will need to copy over the logvar values of the decoder from the original weights for all extant models. I was thinking that since it's a single float scalar we could just update the `config.json` for existing models. For new models, it should just init to `0`.

Longer term it would be good to support the actual discriminators and their weights for the adversarial loss, but I think that should be a separate model. The `logvar` scalar is basically part of the decoder used to scale loss and is trained alongside the decoder.

**Comment by patrickvonplaten** [comment]
@AmericanPresidentJimmyCarter can you explain a bit more in-depth how the `logvar` paramater is necessary for training?

**Comment by AmericanPresidentJimmyCarter** [comment]
When training the model, you scale the reconstruction loss by `logvar`, that is it is the log of the variance of the reconstruction loss.

`nll_loss = rec_loss / torch.exp(self.logvar) + self.logvar`

It allows the training process to scale the reconstruction loss as necessary, but it is fixed during inference. The scaling of the reconstruction loss by the log variance is not part of the discriminator.

**Comment by patrickvonplaten** [comment]
And the `logvar` is learned during training? Or is it fixed? Because if it's fixed I'm not sure we need to add it the model necessarily?

**Comment by AmericanPresidentJimmyCarter** [comment]
It is learned.
